source/fundamentals/builders/filters.txt

```diff
@@ -11,3 +11,571 @@ Filters Builders
+   :copyable: false
+   
+   {"_id": 2, "color": "purple", "qty": 10, "vendor": ["C", "D"]}
+   {"_id": 3, "color": "blue", "qty": 8, "vendor": ["B", "A"]}
+   {"_id": 5, "color": "yellow", "qty": 11, "vendor": ["A", "B"]}
+   {"_id": 7, "color": "green", "qty": 8, "vendor": ["B", "C"]}
+
+The following example creates a filter that matches all documents in
+the ``paint_purchases`` collection because the predicate is empty: 
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Filters.java
+   :language: java
+   :dedent:
+   :start-after: begin emptyComparison
+   :end-before: end emptyComparison
+
+The JSON output renders as the following:
+
+.. code-block:: json
+   :copyable: false
+   
+   { }
+
+The output of the above code snippet should look something like this:
+   
+.. code-block:: json
+   :copyable: false
+      
+   {"_id": 1, "color": "red", "qty": 5, "vendor": ["A"]}
+   {"_id": 2, "color": "purple", "qty": 10, "vendor": ["C", "D"]}
+   {"_id": 3, "color": "blue", "qty": 8, "vendor": ["B", "A"]}
+   {"_id": 4, "color": "white", "qty": 6, "vendor": ["D"]}
+   {"_id": 5, "color": "yellow", "qty": 11, "vendor": ["A", "B"]}
+   {"_id": 6, "color": "pink", "qty": 5, "vendor": ["C"]}
+   {"_id": 7, "color": "green", "qty": 8,"vendor": ["B", "C"]}
+   {"_id": 8, "color": "orange", "qty": 7, "vendor": ["A", "D"]}
+
+.. _logical:
+
+Logical
+-------
+The logical operator looks at the conditions of the specified filters.
+
+The logical operator methods include:
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Logical Method
+     - Matches
+      
+   * - :java-core-api:`and() <com/mongodb/client/model/Filters.html#and(java.lang.Iterable)>`
+     - documents with the conditions of all the filters. This operator joins filters with a logical ``AND``.
+
+   * - :java-core-api:`or() <com/mongodb/client/model/Filters.html#or(org.bson.conversions.Bson...)>`
+     - documents with the conditions of either filter. This operator joins filters with a logical ``OR``.
+      
+   * - :java-core-api:`not() <com/mongodb/client/model/Filters.html#not(org.bson.conversions.Bson)>`
+     - documents that do not match the filter.
+     
+   * - :java-core-api:`nor() <com/mongodb/client/model/Filters.html#nor(org.bson.conversions.Bson...)>`
+     - documents that fail to match both filters. This operator joins filters with a logical ``NOR``.
+
+The following example creates a filter that matches documents where
+the value of the ``qty`` field is greater than "8" or the value
+of the ``color`` field equals "pink" in the ``paint_purchases`` collection: 
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Filters.java
+   :language: java
+   :dedent:
+   :start-after: begin orComparison
+   :end-before: end orComparison
+   
+The JSON output renders as the following:
+
+.. code-block:: json
+   :copyable: false
+   
+   { 
+      "$or" :
+      [
+         "qty" : { "$gt" : 8 }, 
+         "color" : "pink"
+      ]       
+   }
```

Comment:
Nitpick: Unless there's a reason not to, can you make this more compact? Maybe the curly braces could be on the same line and the square braces could be the trailing character on the previous lines.